okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java

```diff
@@ -73,24 +72,25 @@ public ProcessModuleHandle(Vertx vertx, LaunchDescriptor desc, String id,
   }
 
   @Override
-  public void start(Handler<AsyncResult<Void>> startFuture) {
-    if (port > 0) {
+  public Future<Void> start() {
+    Promise<Void> promise = Promise.promise();
+    if (port == 0) {
+      promise.complete();
+    } else {
       // fail if port is already in use
       NetClientOptions options = new NetClientOptions().setConnectTimeout(200);
       NetClient c = vertx.createNetClient(options);
-      c.connect(port, "localhost", res -> {
+      c.connect(port, "localhost").onComplete(res -> {
         if (res.succeeded()) {
           NetSocket socket = res.result();
           socket.close();
-          startFuture.handle(Future.failedFuture(
-              messages.getMessage("11502", Integer.toString(port))));
+          promise.fail(messages.getMessage("11502", Integer.toString(port)));
         } else {
-          start2(startFuture);
+          promise.complete();
         }
       });
-    } else {
-      start2(startFuture);
     }
+    return promise.future().compose(x -> start2());
```

Comment:
No need for the promise:

    if (port == 0) {
      return start2();
    }
    // fail if port is already in use
    NetClientOptions options = new NetClientOptions().setConnectTimeout(200);
    NetClient c = vertx.createNetClient(options);
    return c.connect(port, "localhost").compose(socket -> {
      socket.close();
      return Future.failedFuture(messages.getMessage("11502", Integer.toString(port)));
    }, fail -> start2());

Reply:
oh .. the 2nd argument to compose is useful for "negating".